relayer/strategies.go

```diff
@@ -37,9 +67,16 @@ func StartRelayer(
 	memo string,
 	processorType string,
 	initialBlockHistory uint64,
+	pathName string,
+	prometheusListen string,
```

Comment:
I've written many Go tests involving http servers. We can have a significantly better testing experience if we can pass in the straight `net.Listener` here. 

We can bind to an anonymous port in test, and it's trivial to discover the bound port when we pass in the `net.Listener`. This makes it even easier to run tests in parallel since we can rely on the OS to not pick conflicting ports.

It also allows the CLI to attempt to bind to the port very early, so that we don't waste time initializing other things before finding out someone else bound to our desired port and having to fail.

Reply:
moved this out to be the responsibility of the caller, and made the method take a `net.Listener` similar to the debug server.